install_script.sh.template

```diff
@@ -445,15 +445,44 @@ if [ -n "$DD_AGENT_FLAVOR" ]; then
     agent_flavor=$DD_AGENT_FLAVOR #Eg: datadog-iot-agent
 fi
 
-declare -A flavor_to_readable
-flavor_to_readable=(
-    ["datadog-agent"]="Datadog Agent"
-    ["datadog-iot-agent"]="Datadog IoT Agent"
-    ["datadog-dogstatsd"]="Datadog Dogstatsd"
-    ["datadog-fips-proxy"]="Datadog FIPS Proxy"
-    ["datadog-heroku-agent"]="Datadog Heroku Agent"
-)
-nice_flavor=${flavor_to_readable[$agent_flavor]}
+# Emulate hashmap with simple switch case
+getMapData() {
+
+    if [ "$1" = "flavor_to_readable" ]; then
+        case "$2" in
+            "datadog_agent")
+                DATA="Datadog Agent"
+                ;;
+             "datadog_iot_agent")
+                DATA="Datadog IoT Agent"
+                ;;
+             "datadog_dogstatsd")
+                DATA="Datadog Dogstatsd"
+                ;;
+             "datadog_fips_proxy")
+                DATA="Datadog FIPS Proxy"
+                ;;
+             "datadog_heroku_agent")
+                DATA="Datadog Heroku Agent"
+                ;;
+              *)
+                DATA="Unknown"
+                ;;
+         esac
+    elif [ "$1" = "flavor_to_system_service" ] && [ "$2" = "datadog_dogstatsd" ]; then
```

Comment:
I'm not sure I would have set these additional maps (`flavor_to_system_service`, `flavor_to_etcdir` and `flavor_to_config`) in the same method: theses "maps" had only one data, and where used only once, so having the `if` block close to the usage would make more sense to me (rather than mixing this in a method with other purpose)
Then your generic "getMapData" would only focus on "flavor_to_readable", wdyt?

Reply:
I think even with only one data inside of the Map, it was done this way for scalability purposes (otherwise we'd have just put a constant string value instead already). If I remove them outside of the generic "getMapData" then we'll loose in that regards. I think it's easy to maintain and still allow to add new flavor if needed wdyt ?

Reply:
by scalability you mean "possibility to add more values in the future"? If yes, I would prefer to stick with what we need (so yes constant values when there is no mapping to do), and introduce the map when we have several data to test.